state/convert.go

```diff
@@ -441,7 +516,38 @@ func verifyBlockHash(spork *config.Spork, hash []byte, height uint64, hdr *entit
 		receiptIDs = append(receiptIDs, receipt.ID())
 	}
 	receiptHash := flow.MerkleRoot(receiptIDs...)
-	resultIDs := []flow.Identifier{}
+	var resultIDs []flow.Identifier
+	if spork.Version >= 5 {
+		// for hashing versions V5 (mainnet22) and above
+		// todo: remove this double check after mainnet23 spork, and have just one call to convertExecutionResult
+		var resultIDsV5 []flow.Identifier
+		for _, src := range block.ExecutionResultList {
+			exec, ok := convertExecutionResult(hash, height, src)
+			if ok {
+				resultIDs = append(resultIDs, deriveExecutionResult(spork, exec))
+			}
+			execV5, okV5 := convertExecutionResultV5(hash, height, src)
+			if okV5 {
+				resultIDsV5 = append(resultIDsV5, deriveExecutionResultV5(execV5))
+			}
+			if !okV5 && !ok {
+				// couldn't convert either way, go by regular code path
+				return false
+			}
+		}
+		resultHash := flow.MerkleRoot(resultIDs...)
+		resultHashV5 := flow.MerkleRoot(resultIDsV5...)
+		payloadHash := flow.ConcatSum(collectionHash, sealHash, receiptHash, resultHash)
+		payloadHashV5 := flow.ConcatSum(collectionHash, sealHash, receiptHash, resultHashV5)
+		if payloadHash != xhdr.PayloadHash && payloadHashV5 != xhdr.PayloadHash {
+			log.Errorf(
+				"Mismatching payload hash for block %x at height %d: expected %x, got %x for version 6 and %x for Versions before it",
+				hash, height, xhdr.PayloadHash[:], payloadHash[:], payloadHashV5[:],
+			)
+			return false
+		}
+		return true
```

Comment:
```suggestion
```

Reply:
changed this by removing the `if` statement